pkg/odo/cli/component/create.go

```diff
@@ -115,9 +119,10 @@ odo catalog list components
 %[1]s nodejs --app myapp --project myproject`)
 
 // NewCreateOptions returns new instance of CreateOptions
-func NewCreateOptions(prjClient project.Client) *CreateOptions {
+func NewCreateOptions(prjClient project.Client, prefClient preference.Client) *CreateOptions {
 	return &CreateOptions{
-		PushOptions: NewPushOptions(prjClient),
+		PushOptions: NewPushOptions(prjClient, prefClient),
+		prefClient:  prefClient,
```

Comment:
I don't think you will need a separate prefClient here. CreateOptions 'inherits' PushOptions, thereby getting access to prefClient.